### PR TITLE
DBZ-3515 Execute periodic TX commits

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-3515

### CODE REVIEW

**Comment by jpechane** [comment]
Test failure is unrelated and is already logged as instable test

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java [hunk 1]
```diff
@@ -287,6 +294,16 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
         }
     }
 
+    private void commitTransaction() throws SQLException {
+        // When reading from read-only Always On replica the default and only transaction isolation
+        // is snapshot. This means that CDC metadata are not visible for long-running transactions.
+        // It is thus necessary to restart the transaction before every read.
+        // For R/W database it is improtant to execute regular commits to maintain the size of TempDB
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java:301]:
```suggestion
        // For R/W database it is important to execute regular commits to maintain the size of TempDB
```

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Test failure unrelated, tracked via https://issues.redhat.com/browse/DBZ-3228 (a worthy goal for fixing btw.)

**Comment by jpechane** [comment]
@gunnarmorling Please next time add the DBZ- prefix to the suggestions commit

**Pull request merged**